src/lib/bootloader/mbr_update.rb

```diff
@@ -183,8 +156,8 @@ def activatable_partitions(disk)
       disk.partitions.reject { |p| p.id.is?(:swap, :bios_boot) }
     end
 
-    def extended_partition(disk)
-      part = activatable_partitions(disk).find { |p| p.type == Y2Storage::PartitionType::EXTENDED }
+    def extended_partition(partition)
+      part = partition.siblings.find { |p| p.type.is?(:extended) }
```

Comment:
When using `partition.siblings` take into account the the devicegraph looks like this if there are logical partitions:
```
          sda
  _____________________
 |       |      |      |
sda1    sda2   sda3   sda4
                      ____        
                     |    |
                   sda5  sda6
```
So `sda5.siblings` will only return `sda6`, not all the partitions. In the same way, `sda1.siblings` will not include sda5 nor sda6.